src/__experimental__/components/decimal/decimal.component.js

```diff
@@ -0,0 +1,123 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Textbox from '../textbox';
+import I18nHelper from '../../../utils/helpers/i18n';
+import Logger from '../../../utils/logger';
+
+class Decimal extends React.Component {
+  input = null;
+
+  // Create ref to document for tests
+  _document = document;
+
+  formatValue = () => {
+    const { value } = this.props;
+    const { input } = this;
+
+    // Return unformatted value if component has not mounted
+    if (!input) {
+      return value;
+    }
+    // Return unformatted value if input is being used
+    if (input && this._document.activeElement === input.current) {
+      return value;
+    }
+
+    // Only format value if input is not active
+    // Strip delimiters otherwise formatDecimal Helper goes nuts
+    const format = I18nHelper.format();
+    const delimiter = `\\${format.delimiter}`;
+    const removeDelimiter = new RegExp(`[${delimiter}]*`, 'g');
+    const noDelimiters = value.replace(removeDelimiter, '');
+
+    return I18nHelper.formatDecimal(
+      noDelimiters,
+      this.validatePrecision()
+    );
+  }
+
+  validatePrecision = () => {
+    const { precision } = this.props;
+
+    if (precision > 15) {
+      Logger.warn('Precision cannot be greater than 15');
+      return 15;
+    }
+
+    return precision;
+  }
+
+  testValue = (value) => {
```

Comment:
I think that the name is too general, could you rename this method to be more specific? Maybe `isValidDecimal`?